### PR TITLE
Add attach pool option

### PR DESCRIPTION
**Pull request by pvalena**
Support for attaching to specified subscription pool(s).
Resolves issue #36.

### CODE REVIEW

**Comment by strzibny** [comment]
It misses update to documentation (README).

File: plugins/guests/redhat/cap/subscription_manager.rb [hunk 1]
```diff
@@ -23,11 +23,29 @@ def self.subscription_manager_register(machine, ui)
...
+        # Attach subscription pools
+        def self.attach_pools(machine, pools)
+          if pools
+            command = "subscription-manager attach #{pools_to_options(pools)}"
+            machine.communicate.sudo(command)
+          end
+        end
+
+        # Return pools options for subscription-manager
+        def self.pools_to_options(pools)
```
Comment (strzibny) [inline plugins/guests/redhat/cap/subscription_manager.rb]: This method should be private no?

File: plugins/guests/redhat/cap/subscription_manager.rb [hunk 2]
```diff
@@ -82,13 +100,14 @@ def self.registration_command(command)
 
         # Build additional subscription-manager options based on plugin configuration
         def self.configuration_to_options(config)
-          config.force = true unless config.force
+          config.force = true if config.force.nil?
+          config.pools ||= config.pool
 
           # --auto-attach cannot be used in case of org/activationkey registration
```
Comment (strzibny) [inline plugins/guests/redhat/cap/subscription_manager.rb]: Please update the comment on the changed conditional.

File: plugins/guests/redhat/cap/subscription_manager.rb [hunk 3]
```diff
@@ -23,11 +23,29 @@ def self.subscription_manager_register(machine, ui)
...
             error = String.new
             machine.communicate.sudo(registration_command(command)) do |type, data|
-              error += "#{data}" if type == :stderr
+              error += data if type == :stderr
             end
           rescue Vagrant::Errors::VagrantError
             raise Vagrant::Errors::VagrantError.new, error.strip
           end
+
+          attach_pools(machine, machine.config.registration.pools)
```
Comment (strzibny) [inline plugins/guests/redhat/cap/subscription_manager.rb]: What if the user specifies registration.pool?

**Comment by pvalena** [comment]
Rebased.

**Pull request merged**